Missing proxy config in BaselineMojo?

On my company laptop the m2e build can't download the baseline. That machine has a mandatory proxy, which works fine for everything else in Maven and Tycho (and I can access the same URL in the browser). Since the baseline plugin is relatively new, maybe we are missing the code to setup the proxy for that connection?

I remember you did some similar fix to correctly configure a connection because of not using the Maven proxy settings somewhere else some months ago, but I can't point to the exact commit.

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-baseline-plugin:4.0.0-SNAPSHOT:verify (baseline-check) on project org.eclipse.m2e.archetype.common: Loading baseline repositories https://download.eclipse.org/technology/m2e/releases/latest/ failed! Unable to read repository at https://download.eclipse.org/technology/m2e/releases/latest. ConnectException: UnresolvedAddressException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-baseline-plugin:4.0.0-SNAPSHOT:verify (baseline-check) on project org.eclipse.m2e.archetype.common: Loading baseline repositories https://download.eclipse.org/technology/m2e/releases/latest/ failed!
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Loading baseline repositories https://download.eclipse.org/technology/m2e/releases/latest/ failed!
    at org.eclipse.tycho.baseline.BaselineMojo.loadRepositories (BaselineMojo.java:198)
    at org.eclipse.tycho.baseline.BaselineMojo.execute (BaselineMojo.java:144)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.equinox.p2.core.ProvisionException: Unable to read repository at https://download.eclipse.org/technology/m2e/releases/latest.
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.load (SimpleMetadataRepositoryFactory.java:129)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.factoryLoad (MetadataRepositoryManager.java:63)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:787)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:685)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.loadRepository (MetadataRepositoryManager.java:110)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:65)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:57)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepositor (P2RepositoryManager.java:148)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepository (P2RepositoryManager.java:111)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getCompositeMetadataRepository (P2RepositoryManager.java:117)
    at org.eclipse.tycho.baseline.BaselineMojo.loadRepositories (BaselineMojo.java:193)
    at org.eclipse.tycho.baseline.BaselineMojo.execute (BaselineMojo.java:144)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.net.ConnectException
    at jdk.internal.net.http.HttpClientImpl.send (HttpClientImpl.java:573)
    at jdk.internal.net.http.HttpClientFacade.send (HttpClientFacade.java:123)
    at org.eclipse.tycho.p2maven.transport.Java11HttpTransportFactory$Java11HttpTransport.get (Java11HttpTransportFactory.java:92)
    at org.eclipse.tycho.p2maven.transport.SharedHttpCacheStorage$CacheLine.fetchFile (SharedHttpCacheStorage.java:246)
    at org.eclipse.tycho.p2maven.transport.SharedHttpCacheStorage$2.getCacheFile (SharedHttpCacheStorage.java:135)
    at org.eclipse.tycho.p2maven.transport.HttpTransportProtocolHandler.getFile (HttpTransportProtocolHandler.java:50)
    at org.eclipse.tycho.p2maven.transport.TychoRepositoryTransportCacheManager.createCache (TychoRepositoryTransportCacheManager.java:50)
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.getLocalFile (SimpleMetadataRepositoryFactory.java:69)
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.load (SimpleMetadataRepositoryFactory.java:89)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.factoryLoad (MetadataRepositoryManager.java:63)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:787)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:685)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.loadRepository (MetadataRepositoryManager.java:110)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:65)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:57)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepositor (P2RepositoryManager.java:148)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepository (P2RepositoryManager.java:111)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getCompositeMetadataRepository (P2RepositoryManager.java:117)
    at org.eclipse.tycho.baseline.BaselineMojo.loadRepositories (BaselineMojo.java:193)
    at org.eclipse.tycho.baseline.BaselineMojo.execute (BaselineMojo.java:144)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.net.ConnectException
    at jdk.internal.net.http.common.Utils.toConnectException (Utils.java:1047)
    at jdk.internal.net.http.PlainHttpConnection.connectAsync (PlainHttpConnection.java:198)
    at jdk.internal.net.http.AsyncSSLConnection.connectAsync (AsyncSSLConnection.java:56)
    at jdk.internal.net.http.Http2Connection.createAsync (Http2Connection.java:378)
    at jdk.internal.net.http.Http2ClientImpl.getConnectionFor (Http2ClientImpl.java:128)
    at jdk.internal.net.http.ExchangeImpl.get (ExchangeImpl.java:93)
    at jdk.internal.net.http.Exchange.establishExchange (Exchange.java:343)
    at jdk.internal.net.http.Exchange.responseAsyncImpl0 (Exchange.java:475)
    at jdk.internal.net.http.Exchange.responseAsyncImpl (Exchange.java:380)
    at jdk.internal.net.http.Exchange.responseAsync (Exchange.java:372)
    at jdk.internal.net.http.MultiExchange.responseAsyncImpl (MultiExchange.java:408)
    at jdk.internal.net.http.MultiExchange.lambda$responseAsyncImpl$7 (MultiExchange.java:449)
    at java.util.concurrent.CompletableFuture.uniHandle (CompletableFuture.java:934)
    at java.util.concurrent.CompletableFuture.uniHandleStage (CompletableFuture.java:950)
    at java.util.concurrent.CompletableFuture.handle (CompletableFuture.java:2340)
    at jdk.internal.net.http.MultiExchange.responseAsyncImpl (MultiExchange.java:439)
    at jdk.internal.net.http.MultiExchange.lambda$responseAsync0$2 (MultiExchange.java:341)
    at java.util.concurrent.CompletableFuture$UniCompose.tryFire (CompletableFuture.java:1150)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture$AsyncSupply.run (CompletableFuture.java:1773)
    at jdk.internal.net.http.HttpClientImpl$DelegatingExecutor.execute (HttpClientImpl.java:157)
    at java.util.concurrent.CompletableFuture.completeAsync (CompletableFuture.java:2673)
    at jdk.internal.net.http.MultiExchange.responseAsync (MultiExchange.java:294)
    at jdk.internal.net.http.HttpClientImpl.sendAsync (HttpClientImpl.java:654)
    at jdk.internal.net.http.HttpClientImpl.send (HttpClientImpl.java:552)
    at jdk.internal.net.http.HttpClientFacade.send (HttpClientFacade.java:123)
    at org.eclipse.tycho.p2maven.transport.Java11HttpTransportFactory$Java11HttpTransport.get (Java11HttpTransportFactory.java:92)
    at org.eclipse.tycho.p2maven.transport.SharedHttpCacheStorage$CacheLine.fetchFile (SharedHttpCacheStorage.java:246)
    at org.eclipse.tycho.p2maven.transport.SharedHttpCacheStorage$2.getCacheFile (SharedHttpCacheStorage.java:135)
    at org.eclipse.tycho.p2maven.transport.HttpTransportProtocolHandler.getFile (HttpTransportProtocolHandler.java:50)
    at org.eclipse.tycho.p2maven.transport.TychoRepositoryTransportCacheManager.createCache (TychoRepositoryTransportCacheManager.java:50)
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.getLocalFile (SimpleMetadataRepositoryFactory.java:69)
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.load (SimpleMetadataRepositoryFactory.java:89)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.factoryLoad (MetadataRepositoryManager.java:63)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:787)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:685)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.loadRepository (MetadataRepositoryManager.java:110)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:65)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:57)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepositor (P2RepositoryManager.java:148)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepository (P2RepositoryManager.java:111)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getCompositeMetadataRepository (P2RepositoryManager.java:117)
    at org.eclipse.tycho.baseline.BaselineMojo.loadRepositories (BaselineMojo.java:193)
    at org.eclipse.tycho.baseline.BaselineMojo.execute (BaselineMojo.java:144)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.nio.channels.UnresolvedAddressException
    at sun.nio.ch.Net.checkAddress (Net.java:149)
    at sun.nio.ch.Net.checkAddress (Net.java:157)
    at sun.nio.ch.SocketChannelImpl.checkRemote (SocketChannelImpl.java:816)
    at sun.nio.ch.SocketChannelImpl.connect (SocketChannelImpl.java:839)
    at jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$0 (PlainHttpConnection.java:183)
    at java.security.AccessController.doPrivileged (AccessController.java:569)
    at jdk.internal.net.http.PlainHttpConnection.connectAsync (PlainHttpConnection.java:185)
    at jdk.internal.net.http.AsyncSSLConnection.connectAsync (AsyncSSLConnection.java:56)
    at jdk.internal.net.http.Http2Connection.createAsync (Http2Connection.java:378)
    at jdk.internal.net.http.Http2ClientImpl.getConnectionFor (Http2ClientImpl.java:128)
    at jdk.internal.net.http.ExchangeImpl.get (ExchangeImpl.java:93)
    at jdk.internal.net.http.Exchange.establishExchange (Exchange.java:343)
    at jdk.internal.net.http.Exchange.responseAsyncImpl0 (Exchange.java:475)
    at jdk.internal.net.http.Exchange.responseAsyncImpl (Exchange.java:380)
    at jdk.internal.net.http.Exchange.responseAsync (Exchange.java:372)
    at jdk.internal.net.http.MultiExchange.responseAsyncImpl (MultiExchange.java:408)
    at jdk.internal.net.http.MultiExchange.lambda$responseAsyncImpl$7 (MultiExchange.java:449)
    at java.util.concurrent.CompletableFuture.uniHandle (CompletableFuture.java:934)
    at java.util.concurrent.CompletableFuture.uniHandleStage (CompletableFuture.java:950)
    at java.util.concurrent.CompletableFuture.handle (CompletableFuture.java:2340)
    at jdk.internal.net.http.MultiExchange.responseAsyncImpl (MultiExchange.java:439)
    at jdk.internal.net.http.MultiExchange.lambda$responseAsync0$2 (MultiExchange.java:341)
    at java.util.concurrent.CompletableFuture$UniCompose.tryFire (CompletableFuture.java:1150)
    at java.util.concurrent.CompletableFuture.postComplete (CompletableFuture.java:510)
    at java.util.concurrent.CompletableFuture$AsyncSupply.run (CompletableFuture.java:1773)
    at jdk.internal.net.http.HttpClientImpl$DelegatingExecutor.execute (HttpClientImpl.java:157)
    at java.util.concurrent.CompletableFuture.completeAsync (CompletableFuture.java:2673)
    at jdk.internal.net.http.MultiExchange.responseAsync (MultiExchange.java:294)
    at jdk.internal.net.http.HttpClientImpl.sendAsync (HttpClientImpl.java:654)
    at jdk.internal.net.http.HttpClientImpl.send (HttpClientImpl.java:552)
    at jdk.internal.net.http.HttpClientFacade.send (HttpClientFacade.java:123)
    at org.eclipse.tycho.p2maven.transport.Java11HttpTransportFactory$Java11HttpTransport.get (Java11HttpTransportFactory.java:92)
    at org.eclipse.tycho.p2maven.transport.SharedHttpCacheStorage$CacheLine.fetchFile (SharedHttpCacheStorage.java:246)
    at org.eclipse.tycho.p2maven.transport.SharedHttpCacheStorage$2.getCacheFile (SharedHttpCacheStorage.java:135)
    at org.eclipse.tycho.p2maven.transport.HttpTransportProtocolHandler.getFile (HttpTransportProtocolHandler.java:50)
    at org.eclipse.tycho.p2maven.transport.TychoRepositoryTransportCacheManager.createCache (TychoRepositoryTransportCacheManager.java:50)
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.getLocalFile (SimpleMetadataRepositoryFactory.java:69)
    at org.eclipse.equinox.internal.p2.metadata.repository.SimpleMetadataRepositoryFactory.load (SimpleMetadataRepositoryFactory.java:89)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.factoryLoad (MetadataRepositoryManager.java:63)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:787)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:685)
    at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.loadRepository (MetadataRepositoryManager.java:110)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:65)
    at org.eclipse.tycho.p2maven.transport.RemoteMetadataRepositoryManager.loadRepository (RemoteMetadataRepositoryManager.java:57)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepositor (P2RepositoryManager.java:148)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getMetadataRepository (P2RepositoryManager.java:111)
    at org.eclipse.tycho.p2maven.repository.P2RepositoryManager.getCompositeMetadataRepository (P2RepositoryManager.java:117)
    at org.eclipse.tycho.baseline.BaselineMojo.loadRepositories (BaselineMojo.java:193)
    at org.eclipse.tycho.baseline.BaselineMojo.execute (BaselineMojo.java:144)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :org.eclipse.m2e.archetype.common
```

--- a/p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/helper/ProxyHelper.java
+++ b/p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/helper/ProxyHelper.java
@@ -70,6 +70,9 @@ public class ProxyHelper implements Initializable {
 		if (org.eclipse.aether.repository.Proxy.TYPE_HTTP.equalsIgnoreCase(type)) {
 			return Type.HTTP;
 		}
+		if (org.eclipse.aether.repository.Proxy.TYPE_HTTPS.equalsIgnoreCase(type)) {
+			return Type.HTTP;
+		}
 		return Type.SOCKS;
 	}
 

